build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -173,6 +173,11 @@ function updateReporters(config) {
     config.reporters = ['mocha'];
   }
 
+  if (isCiBuild()) {
+    config.reporters.push('junit');
+    config.junitReporter.outputFile = `result-reports/${config.testType}.xml`;
```

Comment:
Curious: What's the difference between Karma's junit reporter and Mocha's? (I ask because our unit, integration, and e2e tests all use Mocha. Can we try it (and inject a fake test failure) to see if there is a difference in the output and pick the one that does a better job?

Reply:
We use mocha inside of karma which results in mocha being run in the browser. That means that the browsers inability to save files leads to some weird incompatibilities.

This (closed) issue on the mocha project does a pretty good job summing up the problem.
https://github.com/mochajs/mocha/issues/3068

Reply:
Gotcha. There's another thing missing here though. This base class only applies to the unit and integration tests. The E2E tests have their own base class here. It uses Mocha (no Karma).

https://github.com/ampproject/amphtml/blob/f8ba4a462affe59c83ae77c358dc514b121669a8/build-system/tasks/e2e/index.js#L75-L97

To get output for those tests, you'll probably need to use [this workflow](https://circleci.com/docs/2.0/collect-test-data/#mocha-for-nodejs). When this works, we should see tests results here: https://app.circleci.com/pipelines/github/ampproject/amphtml/6299/workflows/1e6545e8-8eef-4078-9faa-f09c97974a63/jobs/89748/tests